declaration variables for different types

I got a linter hit:

```
thelper.go:184:2: declarations should never be cuddled (wsl)
        var log string
        ^
```

for code block:

```
func (h *THelper) Conclude() {                                                 
        var err string                                                         
        var log string                                                         
                                                                               
        if h.Controller == nil {                                               
                goto endConclusion                                             
        }                                                                      
                                                                               
        // process error messages                                              
        err = strings.Join(h.failed, "\n")                                     
                                                                               
        // process log messages                                                
        log = strings.Join(h.log, "\n")                                        
                                                                               
endConclusion:                                                                 
        switch {                                                               
        case h.QuietMode && err == "":                                         
        case h.QuietMode && err != "":                                         
                fallthrough                                                    
        case log != "" && err != "":                                           
                h.Controller.Errorf("\n%v\n%v\n\n\n", err, log)                
        case log != "":                                                        
                h.Controller.Logf("\n%v\n\n\n", log)                           
        case err != "":                                                        
                h.Controller.Errorf("\n%v\n\n\n", err)                         
        }                                                                      
        h.Flush(AllType)                                                       
}
```

Looks fine to me though, as in, group all `var` declaration. What was the initial preference?

--- a/cmd/wsl/main.go
+++ b/cmd/wsl/main.go
@@ -34,6 +34,7 @@ func main() {
 	flag.BoolVar(&config.AllowAssignAndCallCuddle, "allow-assign-and-call", true, "Allow assignments and calls to be cuddled (if using same variable/type)")
 	flag.BoolVar(&config.AllowMultiLineAssignCuddle, "allow-multi-line-assign", true, "Allow cuddling with multi line assignments")
 	flag.BoolVar(&config.AllowCaseTrailingWhitespace, "allow-case-trailing-whitespace", false, "Allow case statements to end with an empty line")
+	flag.BoolVar(&config.AllowCuddleDeclaration, "allow-declarations", false, "Allow declarations to be cuddled")
 
 	flag.Parse()
 

--- a/wsl.go
+++ b/wsl.go
@@ -65,6 +65,13 @@ type Configuration struct {
 	// }
 	AllowCaseTrailingWhitespace bool
 
+	// AllowCuddleDeclaration will allow multiple var/declaration statements to
+	// be cuddled. This defaults to false but setting it to true will enable the
+	// following example:
+	//  var foo bool
+	//  var err error
+	AllowCuddleDeclaration bool
+
 	// AllowCuddleWithCalls is a list of call idents that everything can be
 	// cuddled with. Defaults to calls looking like locks to support a flow like
 	// this:
@@ -362,7 +369,9 @@ func (p *Processor) parseBlockStatements(statements []ast.Stmt) {
 
 			p.addError(t.Pos(), "assignments should only be cuddled with other assignments")
 		case *ast.DeclStmt:
-			p.addError(t.Pos(), "declarations should never be cuddled")
+			if !p.config.AllowCuddleDeclaration {
+				p.addError(t.Pos(), "declarations should never be cuddled")
+			}
 		case *ast.ExprStmt:
 			switch previousStatement.(type) {
 			case *ast.DeclStmt, *ast.ReturnStmt:

--- a/wsl_test.go
+++ b/wsl_test.go
@@ -1188,6 +1188,23 @@ func TestWithConfig(t *testing.T) {
 				AllowCaseTrailingWhitespace: true,
 			},
 		},
+		{
+			description: "allow cuddle var",
+			code: []byte(`package main
+
+			func main() {
+				var t bool
+				var err error
+
+				var t = true
+				if t {
+					fmt.Println("x")
+				}
+			}`),
+			customConfig: &Configuration{
+				AllowCuddleDeclaration: true,
+			},
+		},
 	}
 
 	for _, tc := range cases {

